src/components/complex/Proposal/index.jsx

```diff
@@ -239,6 +243,87 @@ const Proposal = ({
     }
   }, [noData, activeChainId])
 
+  const handleMessageFromIframe = (event) => {
+    // Check the origin of the message for security
+    if (event.origin !== window.location.origin) {
+      console.error('Received an untrusted message from:', event.origin)
+      return
+    }
+
+    // Log or handle the received message
+    if (event.data.type === 'linkClick') {
+      if (event.data.id !== id) return
+      setReadMoreContent(event.data.href)
+      setLocation({ current: event.data.href, old: location.current })
+    }
+  }
+
+  const goBack = () => {
+    setReadMoreContent(location.old)
+    setLocation({ current: location.old, old: url })
+  }
+
+  useEffect(() => {
+    window.addEventListener('message', handleMessageFromIframe)
+
+    return () => {
+      window.removeEventListener('message', handleMessageFromIframe)
+    }
+  }, [])
+
+  useEffect(() => {
+    // // Function to perform search and replace within the iframe
+    const performSearchAndReplace = async () => {
```

Comment:
This function is too complex in my opinion, and needs refactoring to make it smaller and easier to read.